### ISSUE TITLE
Question about the definitions in block sparse attention

### ISSUE THREAD

**Issue by szhengac** [microsoft/DeepSpeed#519]
Hi, I have some question regarding the block sparse attention.

If I understand the description of API correctly, `block` is the block size (i.e., number of tokens in a block) while `num_local_blocks` denotes the number of blocks (`#tokens_per_window = block * num_local_blocks`) in a local window. So no matter which value (`unidirectional` or `bidirectional`) I choose for `attention`, the tokens within a block will attend each other?

**Comment by arashashari** [comment]
Yes, that is correct. Tokens within a block always attend to each other no matter if it is uni/bi-directional. However, if you look at a local window, in the case of unidirectional, you can consider tokens within a block only attend to other tokens in the blocks before them in the same local window. While in case of bidirectional all tokens in the local window (no matter which block they are in) attend to each other.

**Comment by szhengac** [comment]
So if we want to use it for the machine translation, how do we set these
hyper parameters? For machine translation, tokens within a block cannot
attend to each other in decoder.

**Comment by arashashari** [comment]
You can use the attention mask to neutralize it; in such cases, attention mask is of dimension [leading dimensions, S, S] in which S stands for sequence length.

**Comment by szhengac** [comment]
I see. That is indeed one option.
